clang/lib/AST/ASTImporter.cpp

```diff
@@ -3647,15 +3647,19 @@ class IsTypeDeclaredInsideVisitor
 };
 } // namespace
 
-/// This function checks if the function has 'auto' return type that contains
+/// This function checks if the given function has a return type that contains
 /// a reference (in any way) to a declaration inside the same function.
-bool ASTNodeImporter::hasAutoReturnTypeDeclaredInside(FunctionDecl *D) {
+bool ASTNodeImporter::hasReturnTypeDeclaredInside(FunctionDecl *D) {
   QualType FromTy = D->getType();
   const auto *FromFPT = FromTy->getAs<FunctionProtoType>();
   assert(FromFPT && "Must be called on FunctionProtoType");
 
+  bool IsLambdaDefinition = false;
+  if (const auto *MD = dyn_cast<CXXMethodDecl>(D))
+    IsLambdaDefinition = cast<CXXRecordDecl>(MD->getDeclContext())->isLambda();
```

Comment:
Thanks, this works, but if we are going this way, why not also check the lambda was defined without a return type?

Reply:
> Thanks, this works, but if we are going this way, why not also check the lambda was defined without a return type?

Could you please give some direction on how to test **the lambda was defined without a return type**?

Does this mean that we need to check for possible `return` statements within lambda?

Reply:
Or does this mean to test `!FromFPT->hasTrailingReturn()`?

Reply:
No, you don't need to hunt for return statements.

You can just check on the CXXMethodDecl that it's FunctionProtoType has a trailing return (`hasTrailingReturn` method).
I think that should cover it for all cases.

Reply:
Also, this problem of the return type not being wrapped on an AutoType only happens on lambdas in *C++11*.

Reply:
Got it! Thanks for these useful info!